src/state/slices/marketDataSlice/selectors.ts

```diff
@@ -27,18 +22,27 @@ export const selectMarketData = createDeepEqualOutputSelector(
   selectSelectedCurrency,
   (cryptoMarketData, fiatMarketData, selectedCurrency): MarketDataById<AssetId> => {
     const fiatPrice = bnOrZero(fiatMarketData[selectedCurrency]?.price ?? 1) // fallback to USD
-    if (fiatPrice.eq(1)) return cryptoMarketData // don't unnecessarily compute price history for USD
-    return Object.entries(cryptoMarketData).reduce<MarketCapResult>(
-      (acc, [caip19, assetMarketData]) => {
-        if (!assetMarketData) return acc
-        acc[caip19] = {
-          ...assetMarketData,
-          price: bnOrZero(assetMarketData.price).times(fiatPrice).toString(),
-          marketCap: bnOrZero(assetMarketData.marketCap).times(fiatPrice).toString(),
-        }
-        return acc
-      },
-      {},
+    return (
+      Object.entries(cryptoMarketData)
+        // apply fiat conversion
+        .map<[AssetId, MarketData]>(([assetId, assetMarketData]) => {
+          return [
+            assetId,
+            {
+              ...assetMarketData,
+              price: bnOrZero(assetMarketData?.price).times(fiatPrice).toString(),
+              marketCap: bnOrZero(assetMarketData?.marketCap).times(fiatPrice).toString(),
+              volume: bnOrZero(assetMarketData?.volume).times(fiatPrice).toString(),
+              changePercent24Hr: assetMarketData?.changePercent24Hr ?? 0,
+            },
+          ]
+        })
+        // sort by market cap
+        .sort(([, a], [, b]) => (bnOrZero(a.marketCap).lt(bnOrZero(b.marketCap)) ? 1 : -1))
```

Comment:
Nitpick, this is already `bnOrZero().toString()`'d on l.34, can ditch the additional `bnOrZero()` wrapper as we're already in the context of a BigNumber operation